src/main/java/vg/civcraft/mc/civmodcore/world/operations/ChunkOperation.java

```diff
@@ -0,0 +1,19 @@
+package vg.civcraft.mc.civmodcore.world.operations;
+
+import java.util.logging.Logger;
+import org.bukkit.Chunk;
+import org.bukkit.plugin.java.JavaPlugin;
+
+public abstract class ChunkOperation {
+
+	protected final JavaPlugin plugin;
+	protected final Logger logger;
+
+	public ChunkOperation(final JavaPlugin plugin) {
+		this.plugin = plugin;
+		this.logger = plugin.getLogger();
+	}
+
+	public abstract void process(final Chunk chunk);
```

Comment:
If you're not actually using the plugin and logger for anything and merely providing them as convenience for possibly implementation, get rid of them entirely. If someone needs something like this they can still add it, but I doubt they will because they usually have a logger at hand from the surrounding context already. I'd just replace this with a Consumer<Chunk> or have ChunkOperation be an internal wrapper that's put around the Consumer<Chunk> passed in (which allows very nice lambda usage of the method instead of anonymous classes, you can't instanciate abstract classes directly through lambdas).

Reply:
The `plugin` field is accessed by the manager to match operations with plugins, though the logger is just there as a courtesy but can be removed.